scripts/o3de/o3de/download.py

```diff
@@ -111,46 +111,60 @@ def download_o3de_object(object_name: str, default_folder_name: str, dest_path:
     origin_uri = downloadable_object_data['origin_uri']
     parsed_uri = urllib.parse.urlparse(origin_uri)
 
-    download_zip_result = utils.download_zip_file(parsed_uri, download_zip_path, force_overwrite, object_name, download_progress_callback)
-    if download_zip_result != 0:
-        return download_zip_result
-
-    if not validate_downloaded_zip_sha256(downloadable_object_data, download_zip_path, f'{object_type}.json'):
-        logger.error(f'Could not validate zip, deleting {download_zip_path}')
-        os.unlink(download_zip_path)
-        return 1
-
-    if not dest_path:
-        dest_path = manifest.get_registered(default_folder=default_folder_name)
-        dest_path = pathlib.Path(dest_path).resolve()
-        dest_path = dest_path / object_name
+    # If we have a git link then we should clone to the given download path here otherwise download and extract the zip
+    
+    if 'github.com' in parsed_uri.netloc:
+        if not dest_path:
+            dest_path = manifest.get_registered(default_folder=default_folder_name)
+            dest_path = pathlib.Path(dest_path).resolve()
+            dest_path = dest_path / object_name
+        else:
+            dest_path = pathlib.Path(dest_path).resolve()
+        clone_result = utils.clone_git_uri(parsed_uri.geturl(), dest_path)
+        if clone_result:
+            logger.error(f'Could not clone {parsed_uri.geturl()}')
+            return 1
```

Comment:
This part of the logic should be isolated into it's own method, probably apart of a `github_download.py`, which allows us to easily isolate the github specific functionality from the general download functionality.

I suggest an API that can accept a URI and return whether it is a github URI and a function for cloning.
i.e
```python
def is_github_uri(uri : str):
   # ...
def clone_from_git_uri -> bool:
    # ....
```